infrastructure/transform/network.tf

```diff
@@ -0,0 +1,188 @@
+        "Microsoft.Network/virtualNetworks/subnets/unprepareNetworkPolicies/action",
+      ]
+      name = "Microsoft.Databricks/workspaces"
+    }
+  }
+}
+
+resource "azurerm_subnet_network_security_group_association" "databricks_container" {
+  subnet_id                 = azurerm_subnet.databricks_container.id
+  network_security_group_id = azurerm_network_security_group.databricks.id
+}
+
+resource "azurerm_subnet_network_security_group_association" "databricks_host" {
+  subnet_id                 = azurerm_subnet.databricks_host.id
+  network_security_group_id = azurerm_network_security_group.databricks.id
+}
+
+resource "azurerm_private_dns_zone" "databricks" {
+  name                = "privatelink.azuredatabricks.net"
+  resource_group_name = var.core_rg_name
+  tags                = var.tags
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "databricks" {
+  name                  = "vnl-dbks-${var.naming_suffix}"
+  resource_group_name   = var.core_rg_name
+  private_dns_zone_name = azurerm_private_dns_zone.databricks.name
+  virtual_network_id    = data.azurerm_virtual_network.core.id
+  tags                  = var.tags
+}
+
+resource "azurerm_private_endpoint" "databricks_control_plane" {
+  name                = "pe-dbks-cp-${var.naming_suffix}"
+  resource_group_name = var.core_rg_name
+  location            = var.core_rg_location
+  subnet_id           = var.core_subnet_id
+  tags                = var.tags
+
+  private_service_connection {
+    name                           = "private-service-connection-databricks-control-plane-${var.naming_suffix}"
+    private_connection_resource_id = azurerm_databricks_workspace.databricks.id
+    is_manual_connection           = false
+    subresource_names              = ["databricks_ui_api"]
+  }
+
+  private_dns_zone_group {
+    name                 = "private-dns-zone-group-databricks-control-plane-${var.naming_suffix}"
+    private_dns_zone_ids = [azurerm_private_dns_zone.databricks.id]
+  }
+}
+
+resource "azurerm_private_endpoint" "databricks_filesystem" {
+  name                = "pe-dbks-fs-${var.naming_suffix}"
+  resource_group_name = var.core_rg_name
+  location            = var.core_rg_location
+  subnet_id           = var.core_subnet_id
+  tags                = var.tags
+
+  private_service_connection {
+    name                           = "private-service-connection-databricks-filesystem-${var.naming_suffix}"
+    private_connection_resource_id = join("", [azurerm_databricks_workspace.databricks.managed_resource_group_id, "/providers/Microsoft.Storage/storageAccounts/${local.storage_account_name}"])
+    is_manual_connection           = false
+    subresource_names              = ["blob"]
+  }
+
+  private_dns_zone_group {
+    name                 = "private-dns-zone-group-databricks-filesystem-${var.naming_suffix}"
+    private_dns_zone_ids = [data.azurerm_private_dns_zone.blobcore.id]
+  }
+}
```

Comment:
`nit` before @martinpeck sees - no new line at end of file ;)

Reply:
Phew, good spot!